sdk/cosmos/azure_data_cosmos/src/retry_policies/resource_throttle_retry_policy.rs

```diff
@@ -0,0 +1,215 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+use super::{RetryPolicy, ShouldRetryResult};
+use async_trait::async_trait;
+use azure_core::http::{RawResponse, StatusCode};
+use std::sync::atomic::{AtomicUsize, Ordering};
+use std::sync::Arc;
+use std::time::Duration;
+use typespec_client_core::http::Request;
+
+/// Retry policy for handling resource throttling (429 TooManyRequests) errors
+///
+/// This policy implements exponential backoff for 429 status codes, respecting both
+/// maximum retry attempts and cumulative wait time limits. It's designed to handle
+/// Azure Cosmos DB throttling scenarios where the service limits request rates.
+///
+/// # Features
+/// - Exponential backoff with configurable delay factor
+/// - Maximum retry attempt limit
+/// - Cumulative wait time tracking to prevent excessive delays
+/// - Thread-safe retry counter using atomic operations
+///
+/// # Example
+/// ```
+/// use azure_data_cosmos::retry_policies::resource_throttle_retry_policy::ResourceThrottleRetryPolicy;
+///
+/// // Create a policy with 3 max retries, 100 second max wait, and backoff factor of 2
+/// let policy = ResourceThrottleRetryPolicy::new(3, 100, 2);
+/// ```
+#[derive(Debug)]
+pub struct ResourceThrottleRetryPolicy {
+    max_attempt_count: usize,
+    backoff_delay_factor: u32,
+    max_wait_time: Duration,
+    current_attempt_count: AtomicUsize,
+    cumulative_retry_delay: Arc<AtomicUsize>,
```

Comment:
The state management here isn't quite right. The `ResourceThrottleRetryPolicy` is a singleton for the entire application, but most of this state is per-request.

Reply:
Ah, and because all the `RetryPolicy` methods take `&self`, you end up having to use atomic types. That's the Rust compiler giving you a hint that something is a little wrong about how you're handling mutability :).

If a method can mutate fields on `self`, it needs to take it in as `&mut self`. Rust will then guarantee that as long as the caller is using safe Rust (i.e. not inside an `unsafe` block), NO other running code is holding a mutable reference.

I suggest a quick review of [References and Borrowing](https://doc.rust-lang.org/book/ch04-02-references-and-borrowing.html) in the Rust Book! Getting the hang of these is a key part of understanding the magic of Rust :)